### PR TITLE
Apply linting and workflow changes

### PR DESCRIPTION
**Pull request by Searge**
- Adjusted `.editorconfig` for consistent line length and indentation
- Added GitHub Actions workflows for Markdown and Terraform linting
- Updated `.gitignore` to exclude sensitive files and added overrides
- Included Markdown linting configuration and ignore files
- Created documentation files for AWS IAM and VPC modules
- Introduced Terraform linting configuration (`.tflint.hcl`)
- Defined Terraform configurations for development and staging environments
- Added [Infracost](https://dashboard.infracost.io/org/geekops) bot

### CODE REVIEW

**Comment by Searge** [comment]
It looks like the linters have failed because the files are empty.

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  tfsec:
+    name: tfsec
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+    - name: Terraform security scan
+      uses: triat/terraform-security-scan@v3.0.0
+      env:
+        GITHUB_TOKEN: ${{ github.token }}
```
Comment (oleksiihead) [inline .github/workflows/lint.yml:45]: change to 
```
GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment (Searge) [reply .github/workflows/lint.yml:45]: No. That's token from the [github context](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context).

**Comment by Searge** [comment]
I have to implement the linters check when files were changed.

**Comment by Searge** [comment]
> It looks like the linters have failed because the files are empty.

So. Looks like this is true. The last one was failed because of this reason.
We need to configure more complex rules if we need to check terraform in Ci/Cd.

cc: @dereban25, @oleksiihead you can merge request. I'm tired :hand: :microphone:

**Pull request merged**